packages/docprovider/src/yprovider.ts

```diff
@@ -70,69 +49,56 @@ export class WebSocketProvider
     user.changed.connect(userChanged);
   }
 
-  get renameAck(): Promise<boolean> {
-    return this._renameAck.promise;
-  }
-
-  setPath(newPath: string): void {
-    if (newPath !== this._path) {
-      this._path = newPath;
-      const encoder = encoding.createEncoder();
-      this._renameAck = new PromiseDelegate<boolean>();
-      encoding.write(encoder, 127);
-      // writing a utf8 string to the encoder
-      const escapedPath = unescape(
-        encodeURIComponent(
-          this._format + ':' + this._contentType + ':' + newPath
-        )
-      );
-      for (let i = 0; i < escapedPath.length; i++) {
-        encoding.write(
-          encoder,
-          /** @type {number} */ escapedPath.codePointAt(i)!
+  get ready(): Promise<boolean> {
```

Comment:
Every time `.ready` is accessed, it is going to make a request. Is this the behavior we want here or should we be caching the response?

Reply:
You're right, I think we should cache it.